include/swift/Driver/ToolChain.h

```diff
@@ -87,12 +87,12 @@ class ToolChain {
 
   /// Packs together information chosen by toolchains to create jobs.
   struct InvocationInfo {
-    const char *ExecutableName;
+    StringRef ExecutableName;
```

Comment:
If I'm not removing that guarantee, would a comment and some assertions suffice?